nntrainer/layers/time_dist.cpp

```diff
@@ -23,6 +23,63 @@ namespace nntrainer {
 
 const std::string TimeDistLayer::type = "time_dist";
 
+static void reshape(Tensor &m) {
+  TensorDim d = m.getDim();
+  m.reshape({d[2], d[1], d[0], d[3]});
+}
+
+void TimeDistLayer::setPosition() {
+  positions[0] = net_input[0]->getVariableRef().getData();
+  positions[1] = net_input[0]->getGradientRef().getData();
+  positions[2] = net_hidden[0]->getVariableRef().getData();
+  positions[3] = net_hidden[0]->getGradientRef().getData();
+}
+
+void TimeDistLayer::transposeInOut() {
+
+  Tensor &input_ = net_input[0]->getVariableRef();
+  input_.copy(transposeTensor(input_));
```

Comment:
Could you consider using `fill` instead of `copy`? This assures that no allocation happens, also returns error if dimension does not meet.

Reply:
And for below as well :)

Reply:
> Could you consider using `fill` instead of `copy`? This assures that no allocation happens, also returns error if dimension does not meet.

It needs to copy the dimension as well. The dimension of the input_ and trasnposed Tensor are differ.

Reply:
Oh I see.